background.js

```diff
@@ -0,0 +1,471 @@
+	const headers = details.requestHeaders;
+
+	for (let i = 0; i < headers.length; i++) {
+		if (headers[i].name.toLowerCase() !== 'X-TFS-Session'.toLowerCase()) {
+			continue;
+		}
+
+		tabToSessionMap[tabId] = headers[i].value;
+	}
+
+	// self delete this listener
+	chrome.webRequest.onBeforeSendHeaders.removeListener(initialListeners[tabId * 2]);
+	if (!tabToSessionMap[tabId]) {
+		throw 'Could not obtain Azure DevOps Session ID.';
+	}
+}
+
+function onStartListener(details, tabId) {
+	if (!isTestRunnerApiCall(details)) {
+		return;
+	}
+
+	let workItemId;
+	let apiCallUserInfoUri;
+	if (details.url.endsWith(API_ON_PREMISE_CALL_OPEN_TEST_RUNNER)) {
+		workItemId = JSON.parse(String.fromCharCode.apply(null, new Uint8Array(details.requestBody.raw[0].bytes))).workItemIds;
+		apiCallUserInfoUri = details.url.substring(0, details.url.length - API_ON_PREMISE_CALL_OPEN_TEST_RUNNER.length) +
+			'/_api/_common/GetUserProfile?__v=5';
+	} else {
+		workItemId = JSON.parse(String.fromCharCode.apply(null, new Uint8Array(details.requestBody.raw[0].bytes))).context.properties.workItemIds;
+		apiCallUserInfoUri = details.url.substring(0, details.url.length - API_SERVICES_CALL_OPEN_TEST_RUNNER.length) +
+			'/_api/_common/GetUserProfile?__v=5';
+	}
+
+	tabToTestCaseMap[tabId] = workItemId;
+
+	resolveUserNameOfTesterAndTriggerRecordingStart(apiCallUserInfoUri, tabId);
+
+	// self delete this listener
+	chrome.webRequest.onBeforeRequest.removeListener(initialListeners[tabId * 2 + 1]);
+}
+
+/**
+ * This listener is attached to web requests in tabs that are assumed to belong to an Azure DevOps test runner and
+ * parses needed information from update calls (regarding a test run) to Azure DevOps.
+ */
+function testRunUpdateCallListener(details, tabId, tab) {
+	if (!details.url.endsWith(API_CALL_UPDATE_TEST_RUN_SUFFIX)) {
+		return;
+	}
+
+	let updateRequest = JSON.parse(JSON.parse(String.fromCharCode.apply(
+		null, new Uint8Array(details.requestBody.raw[0].bytes))).updateRequests)[0];
+
+	const testCaseId = updateRequest.testCaseResult.testCaseId;
+	const updatedIterationActionResult = getUpdatedIterationActionResult(updateRequest);
+	const testRunId = updateRequest.testRunId;
+
+	tabToAdosTestRunUrlMap[tabId] = tab.url.substring(0, tab.url.indexOf('/_testExecution/')) +
+		'/_testManagement/runs?runId=' + testRunId + '&_a=runCharts';
+
+	if (!testParametersMap[testCaseId]) {
+		testParametersMap[testCaseId] = getParameterDefinitionsPerIteration(updateRequest);
+	}
+
+	const testNameWithParameter = updateRequest.testCaseResult.testCaseTitle.trim();
+
+	if (updatedIterationActionResult.outcome === 12) { // test is paused, other call to Teamscale needed
+		queryTeamscale('pause', details.tabId, testNameWithParameter,
+			testOutcomeToTeamscaleTestExecutionResult(updatedIterationActionResult.outcome));
+	} else {
+		queryTeamscale('update', details.tabId, testNameWithParameter,
+			testOutcomeToTeamscaleTestExecutionResult(updatedIterationActionResult.outcome));
```

Comment:
Here, maybe introducing a new local variable, setting it in if-else block and then calling the `queryTeamscale()` method?
```
let action;
if (updatedIterationActionResult.outcome === 12) {
 action = 'pause';
else {
 action = 'update';
}
queryTeamscale(action, details.tabId, testNameWithParameter,
			testOutcomeToTeamscaleTestExecutionResult(updatedIterationActionResult.outcome));
```

Reply:
refactored as suggested, thanks :-)